File: 4/4.cpp
#include <iostream>
using namespace std;


bool is_palindrome(int n) {
	int nd=n;
	int n_rev=0;
	
	while (nd) {
		n_rev *= 10; 
		n_rev += nd%10;
		nd /= 10;
	}
	if (n == n_rev) 
		return true;
	return false;	
}

int main() {
	const int N = 9;
	const int M = 1;
	
	int j;
	for (int total=2*N; total >= 2*M; total--) {
		for (int i=std::max(M, total - N); i <= std::min(N, total - M); i++) {
			j = total - i;
			cout << i << " " << j << endl;
			if (is_palindrome(i*j)) {
				//cout << i*j << endl;
				//return 0;
			}
		}
	}
	return 0;
}

File: 4/4.py
def is_palindrome(n: int) -> bool:
	nd = n
	n_rev = 0

	while nd:
		n_rev *= 10
		n_rev += nd % 10
		nd //= 10

	if n == n_rev:
		return True
	return False	


N: int = 999;
M: int = 100;

for total in range(2*N, 2*M - 1, -1):
	for i in range(max(M, total - N), min(N, total - M) + 1):
		j = total - i
		
		if is_palindrome(i*j):
			print(i*j)
			exit()
			
